src/utill.rs

```diff
@@ -350,17 +350,6 @@ pub fn parse_field<T: std::str::FromStr>(value: Option<&String>, default: T) ->
     }
 }
 
-/// Function to write data to default toml files
-pub fn write_default_config(path: &PathBuf, toml_data: String) -> std::io::Result<()> {
-    if let Some(parent) = path.parent() {
-        fs::create_dir_all(parent)?;
-    }
-    let mut file = File::create(path)?;
-    file.write_all(toml_data.as_bytes())?;
-    file.flush()?;
-    Ok(())
-}
-
```

Comment:
Change this to `write_toml_to_file` and use that in the directory server. 

Also directly call `create_dir_all`. `if let` is not required here.